Xmc4800 automation board support and MEMC drivers for XMC4xxx

talih0 opened the pull request:
- Includes MEMC drivers for XMC4xxx
- The dev kit has an SDRAM connected via MEMC.

talih0 committed:
drivers: memc: Add xmc4xxx EBU

Add xmc4xxx EBU drivers.

Signed-off-by: Andriy Gelman <andriy.gelman@gmail.com>

talih0 committed:
boards: arm: Add xmc4800 automation board v2

Adds xmc4800 automation board.

Signed-off-by: Andriy Gelman <andriy.gelman@gmail.com>

talih0 committed:
boards: arm: xmc45_relax_kit: Add memory regions to linker

Add memory regions to linker.

Signed-off-by: Andriy Gelman <andriy.gelman@gmail.com>

galak commented on dts/bindings/memory-controllers/infineon,xmc4xxx-ebu.yaml:
```diff
@@ -0,0 +1,216 @@
+      required: true
+      description: Selects the region to configure. All the following properties apply to this region.
+
+    device-type:
```
please come up with a different property name.  `device-type` is a standard devicetree property.

talih0 replied:
I added a infineon prefix to the child bindings

galak commented on dts/bindings/memory-controllers/infineon,xmc4xxx-ebu.yaml:
```diff
@@ -0,0 +1,216 @@
+    region-enable:
+      type: boolean
+      description: |
+        Enables the address space 0x6x000000-. This address space supports code
```
This description doesn't seem finished.

talih0 replied:
I added more details in the description.

galak commented on dts/bindings/memory-controllers/infineon,xmc4xxx-ebu.yaml:
```diff
@@ -0,0 +1,216 @@
+
+    alternate-region-enable:
+      type: boolean
+      description: Enables the non-executable alternate address space 0xax000000-.
```
This description doesn't seem finished.

talih0 replied:
I added more details in the description.

galak reviewed (changes requested):
need to rework the `device-type` property in dts binding.

talih0 commented:
Reopen PR.
Waiting for https://github.com/zephyrproject-rtos/zephyr/pull/50880 for to get merged before rebasing.

talih0 commented:
Ping pls. 
The memc driver was tested on XMC4800 automation board to interface with sd ram (tests/drivers/memc/ram) and on a custom board to interface with FPGA.

talih0 commented:
I'll rebase this when https://github.com/zephyrproject-rtos/zephyr/pull/55435 gets merged

talih0 commented:
@npal-cy, any chance you could have a look at this PR? 
The main part is the EBU driver (MEMC) for external interfaces. I added XMC4800 automation board to test the EBU driver with external SDRAM.

npal-cy commented on dts/arm/infineon/xmc4xxx.dtsi:
```diff
@@ -51,6 +51,14 @@
 
 	soc {
+		ebu: ebu@58008000 {
+		    compatible = "infineon,xmc4xxx-ebu";
```
change space to tab

talih0 commented:
@npal-cy thanks, for your comments. I'll update the PR once the xmc47 relax kit is merged.

talih0 commented on drivers/memc/memc_xmc4xxx.c:
```diff
@@ -0,0 +1,133 @@
+
+	XMC_EBU_Enable(cfg->ebu);
+	cfg->ebu->CLC = cfg->clc;
+	while ((cfg->ebu->CLC & CLC_ACK_Msk) >> 4 != cfg->clc)
```
I don't think a timeout is needed. The same approach is used in XMCLib (xmc_ebu.c, xmc_ebu.h).

talih0 commented on drivers/memc/memc_xmc4xxx.c:
```diff
@@ -0,0 +1,133 @@
+		region_config = &cfg->region_config[i];
+		region_index = region_config->region_index;
+
+		if (region_index >= 4) {
```
I can't see an XMC define, but I'll add one to the file, thanks

talih0 committed:
samples: sensor: sensor_shell: Add overlays for xmc4xxx boards

Adds overlays for xmc45_relax_kit, xmc47_relax_kit and
xmc4800_automation_board_v2 to include the die temperature sensor.

Signed-off-by: Andriy Gelman <andriy.gelman@gmail.com>

talih0 commented:
I rebased to get a fresh CI run

talih0 commented:
@npal-cy, could you re-ack when you have a moment, please? The ci errors are from a different board.
It would be great to get this merged before the freeze for the next releasse

talih0 commented:
@galak, can you another look at this PR pls?

talih0 commented:
@galak, ping, can you have another look pls
